django/contrib/admin/actions.py

```diff
@@ -46,10 +47,15 @@ def delete_selected(modeladmin, request, queryset):
             for obj in queryset:
                 obj_display = force_text(obj)
                 modeladmin.log_deletion(request, obj, obj_display)
-            queryset.delete()
-            modeladmin.message_user(request, _("Successfully deleted %(count)d %(items)s.") % {
-                "count": n, "items": model_ngettext(modeladmin.opts, n)
-            }, messages.SUCCESS)
+            try:
+                queryset.delete()
+                modeladmin.message_user(request, _("Successfully deleted %(count)d %(items)s.") % {
+                    "count": n, "items": model_ngettext(modeladmin.opts, n)
+                }, messages.SUCCESS)
+            except ProtectedError as perr:
+                modeladmin.message_user(request, _("Could not delete %(items)s: %(msg)s.") % {
+                    "items": model_ngettext(modeladmin.opts, n), "msg": perr.args[0]
```

Comment:
Not sure about include `perr.args[0]` in the message -- at least the message should be tested so I can see what it says in the end.